### ISSUE TITLE
can't build `borsh` without `std`

### ISSUE THREAD

**Issue by pmnoxx** [near/borsh-rs#59]
Building, `borsh` with not `std` fails, due to lack of `alloc`

```shell
> cargo build --no-default-features

error: cannot find macro `vec` in this scope
   --> borsh/src/de/mod.rs:283:30
    |
283 |             let mut result = vec![T::deserialize(buf)?];
    |                              ^^^
    |
    = note: consider importing one of these items:
            alloc::vec
            crate::maybestd::vec

error: could not compile `borsh` due to previous error
```

**Comment by austinabell** [comment]
https://github.com/near/borsh-rs/pull/47 added this change, probably an easy fix. Not sure why the CI isn't configured to run on branches

**Comment by tspiteri** [comment]
I think this was fixed in #79; it seems #78 was a duplicate of this issue.
